core/schema/container.go

```diff
@@ -232,12 +232,16 @@ func (s *containerSchema) withAllGPUs(ctx context.Context, parent *core.Containe
 }
 
 type containerWithEntrypointArgs struct {
-	Args []string
+	Args            []string
+	KeepDefaultArgs bool
 }
 
 func (s *containerSchema) withEntrypoint(ctx context.Context, parent *core.Container, args containerWithEntrypointArgs) (*core.Container, error) {
 	return parent.UpdateImageConfig(ctx, func(cfg specs.ImageConfig) specs.ImageConfig {
 		cfg.Entrypoint = args.Args
+		if !args.KeepDefaultArgs {
```

Comment:
Hm, I'm not actually convinced about the new option here - I think I'd happily just change the default in every case. If users want the old behavior, you could call `.Entrypoint()` and then use the result from that to reset `.SetEntrypoint`.

Could be convinced otherwise though :tada:

Reply:
> If users want the old behavior, you could call `.Entrypoint()` and then use the result from that to reset `.SetEntrypoint`.

You mean use `.DefaultArgs()` to reset `.WithDefaultArgs()`.

Yeah, I was also on the fence but thought "why not"? Having to fetch `.DefaultArgs` first forces unlazying. Can't be put it in a `With()` util for example, so it's not as easy to update.

@shykes wdyt, since you suggested it?

Reply:
imo provided `KeepDefaultArgs` is optional it's pretty out-of-the-way for the users who don't care about it and pretty low-maintenance to support for the occasional users who do care about it, so it makes sense to have it

Reply:
Makes sense to keep it IMO. Note that I hate having to pass a context & check for error each time I need to query an individual field, so if you give me a way to do that less, I’m always going to say yes 😛

Reply:
Yeah, sure I'm convinced to keep it :tada: I think I'm just genuinely confused as to when you'd really want the old behavior, but I guess we'll see if people start using the new flag.